x-pack/plugins/ml/common/util/errors/process_errors.ts

```diff
@@ -73,6 +73,14 @@ export const extractErrorProperties = (error: ErrorType): MLErrorObject => {
           error.body.attributes.body.error.caused_by?.caused_by?.reason ||
           error.body.attributes.body.error.caused_by?.reason;
       }
+      if (
+        Array.isArray(error.body.attributes.body.error.root_cause) &&
+        typeof error.body.attributes.body.error.root_cause[0] === 'object' &&
+        error.body.attributes.body.error.root_cause[0]?.script
```

Comment:
maybe this can make use of `isPopulatedObject` like `isPopulatedObject(error.body.attributes.body.error.root_cause[0], ['script'])`. If that works it should allow you to use the attribute inside the `if` without the `?`.

Reply:
Good call! Added in a95e1ac2ba95e516c847283561b01f97b14b5503